internal/resource/configvalidators/hostname_or_ip_validator.go

```diff
@@ -0,0 +1,43 @@
+package configvalidators
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var _ validator.String = &hostnameOrIpValidator{}
+
+type hostnameOrIpValidator struct{}
+
+func (v hostnameOrIpValidator) Description(ctx context.Context) string {
+	return "Invalid hostname or IP, Please provide a valid hostname or IP address"
```

Comment:
The descriptions of the validators should describe what they are validating. This sounds more like an error message.

Reply:
added more validator-like descriptions rather than error-like ones